music21/scale/__init__.py

```diff
@@ -872,6 +875,57 @@ def buildNetwork(self):
             'interval': interval.Interval('a1')
         }
 
+    def getRelativeMajor(self):
+        return MajorScale(self.tonic)
+
+    def getRelativeMinor(self):
+        return MinorScale(self.tonic)
```

Comment:
Can you say a bit more about the problem you are trying to solve here?

Works for me without any changes:

```
>>> from music21 import scale
>>> gm = scale.MinorScale('g')
>>> gm.getRelativeMajor()
<music21.scale.MajorScale B- major>
```

Also, at a glance the implementation looks wrong (it's getting parallel scales).

Reply:
Indeed, the natural minor scale was already able to use the relative minor/major functionality, but the harmonic/melodic/jazz minor scales could not. With my changes, your test could work on these other scales as well:

`>>> from music21 import scale`
`>>> gm = scale.HarmonicMinorScale('g')`
`>>> gm.getRelativeMajor()`
`<music21.scale.MajorScale B- major>`

Although it does look strange how the functions use self.tonic rather than self.relativeMinorDegree/self.relativeMajorDegree, that is what I saw other similar scales using to implement the relative key functionality. Nevertheless, the output to the user seems to correctly display relative major and minor scales unless I am missing something.

Reply:
Okay, got it now. Still, I don't think we need to define the functions here on the subclasses, they're already defined on `DiatonicScale`. Actually, now that I look, these are methods on ConcreteScales, so we definitely shouldn't define them on AbstractScales.

Speaking of which, it looks like the (untested) implementation doesn't quite work:

PR:
```
>>> from music21 import *
>>> ms = scale.MelodicMinorScale()
>>> ms.getRelativeMajor()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jwalls/music21/music21/scale/__init__.py", line 2660, in getRelativeMajor
    return MajorScale(self.pitchFromDegree(self.abstract.relativeMajorDegree))
  File "/Users/jwalls/music21/music21/scale/__init__.py", line 1742, in pitchFromDegree
    post = self._abstract.getPitchFromNodeDegree(
  File "/Users/jwalls/music21/music21/scale/__init__.py", line 487, in getPitchFromNodeDegree
    post = self._net.getPitchFromNodeDegree(
  File "/Users/jwalls/music21/music21/scale/intervalNetwork.py", line 2619, in getPitchFromNodeDegree
    realizedPitch, realizedNode = self.realize(
  File "/Users/jwalls/music21/music21/scale/intervalNetwork.py", line 1714, in realize
    raise IntervalNetworkException('pitchReference cannot be None')
music21.scale.intervalNetwork.IntervalNetworkException: pitchReference cannot be None
```

Reply:
In fact, that's probably why the new lines are unexercised on the Coveralls report.

Reply:
right should not be on abstract scales and also returns the parallel, not relative.

Reply:
In my latest commit, while addressing this feedback I removed the individual abstract minor scale classes so that all minor scales would instead be implemented directly within the AbstractDiatonicScale class. In making this change, I hope to increase consistency and user experience slightly, but recognize that there could be other concerns I am not aware of; as such, if y'all think I should leave the abstract harmonic/melodic/jazz minor classes, I can easily do so in a future commit.